File: 03_Data_Structures_&_Counting/14425_String_Set_Membership/Answer14425.kt
fun main(){
  val (n, m) = readln().split(" ").map {it.toInt()}
  val strs = mutableListOf<String>()
  repeat(n){strs.add(readln())};
  val strsSet = strs.toSet()
  var resultInt = 0
  for(i in n until n + m) if(strsSet.contains(readln())) resultInt++
  print(resultInt)
}

File: 03_Data_Structures_&_Counting/14425_String_Set_Membership/Answer14425.rs
use std::collections::HashSet;
use std::io::stdin;

fn main(){
  let mut s = String::new();
  stdin().read_line(&mut s).unwrap();
  let nums:Vec<usize> = s.trim().split_whitespace().map(|x| x.parse().unwrap()).collect();
  let (n, m) = (nums[0], nums[1]);
  let strs:Vec<String> = (0 .. n).map(|_|{
    let mut s = String::new();
    stdin().read_line(&mut s).unwrap();
    s.trim().to_string()}).collect();
  let strs_set:HashSet<String> = strs.iter().cloned().collect();
  let mut result_int = 0;
  for _ in n .. n + m {
    let mut s = String::new();
    stdin().read_line(&mut s).unwrap();
    if strs_set.contains(s.trim()) {result_int += 1;}
  }
  print!("{}", result_int);
}